### PR TITLE
logger: refactor, remove dead code

### PR DESCRIPTION
**Pull request by mroutis**
This PR includes some modifications that I did way back but never pushed them, some problems where already addressed by #3290 

* Remove dead code
* Rename _walk_exc to _causes, simplify it
* Split up stack trace and remove extra \n
* Import traceback only when needed
* Add test `exc_info` in other levels

### CODE REVIEW

**Comment by mroutis** [comment]
uh, thought that I rebased master already...

File: dvc/logger.py [hunk 1]
```diff
@@ -58,76 +56,55 @@ class ColorFormatter(logging.Formatter):
...
-        elif message:
-            description = "{message}"
-
-        return description.format(message=message, exception=exception)
-
-    def _walk_exc(self, exc_info):
-        exc = exc_info[1]
+    @property
+    def _verbose(self):
+        return logging.getLogger("dvc").getEffectiveLevel() == logging.DEBUG
```
Comment (efiop) [inline dvc/logger.py]: Doesn't look like this should be property :)

File: dvc/logger.py [hunk 2]
```diff
@@ -58,76 +56,55 @@ class ColorFormatter(logging.Formatter):
...
 
-        exc_list = [str(exc)]
-
-        while hasattr(exc, "__cause__") and exc.__cause__:
-            exc_list.append(str(exc.__cause__))
+    def _causes(self, exc):
```
Comment (efiop) [inline dvc/logger.py]: staticmethod?

File: dvc/logger.py [hunk 3]
```diff
@@ -58,76 +56,55 @@ class ColorFormatter(logging.Formatter):
...
-        tb_only = getattr(record, "tb_only", False)
-
-        if not record.exc_info:
-            return (None, "")
-
-        exc_list = self._walk_exc(record.exc_info)
-        tb = traceback.format_exception(*record.exc_info)
-
-        exception = None if tb_only else ": ".join(exc_list)
+    def _stack_trace(self, exc_info):
```
Comment (efiop) [inline dvc/logger.py]: staticmethod?

File: dvc/logger.py [hunk 4]
```diff
@@ -58,76 +56,55 @@ class ColorFormatter(logging.Formatter):
...
+
+        if record.exc_info:
+            if getattr(record, "tb_only", False):
+                cause = ""
+            else:
+                cause = ": ".join(self._causes(record.exc_info[1]))
+
+            msg = "{message}{separator}{cause}".format(
+                message=msg or "",
+                separator=" - " if record.msg and cause else "",
```
Comment (efiop) [inline dvc/logger.py]:
```suggestion
                separator=" - " if msg and cause else "",
```
?

**Comment by mroutis** [comment]
Thanks, @efiop , changes addressed, would you mind reviewing it again?

**Pull request merged**